### ISSUE TITLE
Running Cardboard in Unity  on iOS as a debug build (or with Metal Validation on) causes a crash

### ISSUE THREAD

**Issue by Moncader** [googlevr/cardboard#309]
### SPECIFIC ISSUE ENCOUNTERED
Following the exact method to use the Cardboard SDK for Unity on https://developers.google.com/cardboard/develop/unity/quickstart
results in your app crashing 100% of the time in a debug build with default Unity settings.

By default, Unity has the Metal Validation API turned on by default. And during development of an application,
we have this on for debugging purposes as well and is a required feature for app development.

When this is turned on and you start up cardboard, on the first frame rendered, the below Metal validation error occurs and the app will crash.

```
-[MTLDebugRenderCommandEncoder validateFramebufferWithRenderPipelineState:], line 1288: error 'Framebuffer With Render Pipeline State Validation
For color attachment 0, the render pipeline's pixelFormat (MTLPixelFormatRGBA8Unorm) does not match the framebuffer's pixelFormat (MTLPixelFormatBGRA8Unorm).
```

https://github.com/googlevr/cardboard/blob/master/sdk/unity/xr_unity_plugin/metal_renderer.mm#L363
Here, you're preparing a RGBA framebuffer but 
https://github.com/googlevr/cardboard/blob/master/sdk/unity/xr_unity_plugin/metal_renderer.mm#L120
here you're creating a color attachment in BGRA format.

This breaks Metal Validation and simply changing the metal_renderer.mm#L363 from RGBA to BGRA fixes the issue, at least on my end.

### SMARTPHONE (please complete the following information):
- Device: Tested on iPhone 12 Max
- OS
  - [ ] Android
  - [x] iOS
- OS version [e.g. 12]
I've tested on iOS 14.3 and 15.1

### STEPS TO REPRODUCE THE ISSUE
1. Follow all the instructions for iOS and run the sample app for Unity in debug mode.

### EXPECTED BEHAVIOR
No crash

### VERSIONS USED
Cardboard XR Plugin 1.7.0, 1.8.0 and master branch

**If you are using Cardboard XR Plugin:**
- What version of Unity are you using?
Tried 2020.3.19f1 up to 2021.2.0f1

- Select in which graphics API you can reproduce the bug.
   - [ ] OpenGL ES2
   - [ ] OpenGL ES3
   - [x] Metal

**Comment by jballoffet** [comment]
Thanks @Moncader for reaching out to us! I confirm I'm able to reproduce it locally. It would be great if you could send a PR with the proposed changes so as to speed up the fix. In case you'd like to do so, please feel free to do so, just read CONTRIBUTING.md and follow the process. Thanks!

**Comment by Moncader** [comment]
@jballoffet 
I could send a PR, but I'm no expert in Metal and don't know the consequences of changing this this way.

Is there a reason it was originally coded this way?

Just randomly changing colour formats doesn't sound like a safe way to fix something...
If there is proof that this could be a safe fix then I'd be happy to create a PR.

**Comment by Moncader** [comment]
@jballoffet 
I've opened a PR #312 that fixes this on my end.

**Comment by jballoffet** [comment]
Your proposed changes have been already internally merged and will be included in the next release. Thanks for contributing to Cardboard!

**Comment by xinyunh0929** [comment]
This was fixed in v1.11.0
